src/hal/src/pso/mod.rs

```diff
@@ -151,26 +151,53 @@ impl fmt::Display for Stage {
 }
 
 /// Shader entry point.
-#[derive(Debug, Copy)]
-pub struct EntryPoint<'a, B: Backend> {
+#[derive(Debug)]
+pub struct EntryPoint<'a, B: Backend, S = RawSpecialization<'a>> {
     /// Entry point name.
     pub entry: &'a str,
     /// Shader module reference.
     pub module: &'a B::ShaderModule,
     /// Specialization.
-    pub specialization: Specialization<'a>,
+    pub specialization: S,
 }
 
-impl<'a, B: Backend> Clone for EntryPoint<'a, B> {
+impl<'a, B, S> EntryPoint<'a, B, S>
+where
+    B: Backend,
+    S: Specialization,
+{
+    /// Convert `specialization` of the `EntryPoint` into raw type
+    /// acceptable by pipeline creation functions.
+    pub fn as_raw(&'a self) -> EntryPoint<'a, B> {
```

Comment:
The convention of `as_something()` in Rust is to borrow and return a reference. In this case, let's have `to_raw()`

Reply:
Okey. But this changes to the way it was originally.